src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs

```diff
@@ -813,13 +829,11 @@ public void AddEntity(EntityId entityId, IActorRef @ref)
                     _remembering.Remove(entityId);
             }
 
-            public IActorRef Entity(EntityId entityId)
+            public IActorRef? Entity(EntityId entityId)
             {
-                if (_entities.TryGetValue(entityId, out var state))
-                {
-                    if (state is WithRef wr)
-                        return wr.Ref;
-                }
+                if (!_entities.TryGetValue(entityId, out var state)) return null;
+                if (state is WithRef wr)
```

Comment:
NIT:
Not sure whether this would be better:

```
if (_entities.TryGetValue(entityId, out var state) && state is WithRef wr)
{
  return wr.Ref;
}
else
{
  return null;
}
```

But figured it may be worth the ask. If nothing else it lowers dependence on compiler/jitter for epilog.